src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -925,16 +925,23 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger errorLogger, Cancellat
                         additionalFileAnalyzerOptions);
                 }
 
-                Diagnostics.AnalyzerOptions analyzerOptions = CreateAnalyzerOptions(
+                AnalyzerOptions analyzerOptions = CreateAnalyzerOptions(
                     additionalTextFiles, analyzerConfigProvider);
 
+                // Always filter out 'Hidden' analyzer diagnostics in build.
+                // Filter out 'Info' analyzer diagnostics if they are not required to be logged in errorlog.
```

Comment:
Can you change this comment to say a bit about "why"? As written it mostly says "what". 

Think it would be good to mention that we're filtering these out because they don't appear in the output of the build hence running them is just spinning CPU cycles for no gain.

Reply:
Sure, I'll improve the comment.